pgjdbc/src/main/java/org/postgresql/jdbc/PgStatement.java

```diff
@@ -736,7 +736,7 @@ protected BatchResultHandler createBatchHandler(Query[] queries,
         // If executing the same query twice in a batch, make sure the statement
         // is server-prepared. In other words, "oneshot" only if the query is one in the batch
         // or the queries are different
-        && isOneShotQuery(null)) {
+        || isOneShotQuery(null)) {
       flags |= QueryExecutor.QUERY_ONESHOT;
```

Comment:
I don't think this change has the intended effect.
The first check in isOneShotQuery(CachedQuery cachedQuery) is "if (cachedQuery == null) { return true; }".
In other words, the entire "if (!sameQueryAhead || isOneShotQuery(null)) { flags |= QueryExecutor.QUERY_ONESHOT; } else { ... }" block is equivalent to an unconditional "flags |= QueryExecutor.QUERY_ONESHOT;" statement. The else block will never execute now.
Also, the original code is misleading because "if (!sameQueryAhead && isOneShotQuery(null))" is equivalent to "if (!sameQueryAhead && true)".
Note: If this change is not causing a test failure for queries that require "flags |= QueryExecutor.QUERY_FORCE_DESCRIBE_PORTAL", then there is a coverage gap that should be filled or there are no queries that need that flag set or the flag gets set somewhere else.